src/containers/sprite-selector-item.jsx

```diff
@@ -34,6 +39,34 @@ class SpriteSelectorItem extends React.Component {
             'handleMouseMove',
             'handleMouseUp'
         ]);
+        this.svgRenderer = new SVGRenderer();
+        // Asset ID of the SVG currently in SVGRenderer
+        this.svgRendererAssetId = null;
+    }
+    getCostumeUrl () {
+        if (this.props.costumeURL) return this.props.costumeURL;
+        if (!this.props.assetId) return null;
+
+        const storage = this.props.vm.runtime.storage;
+        const asset = storage.get(this.props.assetId);
+        // If the SVG refers to fonts, they must be inlined in order to display correctly in the img tag.
+        // Avoid parsing the SVG when possible, since it's expensive.
+        if (asset.assetType === storage.AssetType.ImageVector) {
+            const svgString = this.props.vm.runtime.storage.get(this.props.assetId).decodeText();
+            if (svgString.match(HAS_FONT_REGEXP)) {
+                // If the asset ID has not changed, no need to re-parse
+                if (this.svgRendererAssetId === this.props.assetId) {
+                    return this.cachedUrl;
+                }
```

Comment:
Should this check move up above `const svgString = ...` to save a bit more time?

Reply:
The svgRendererAssetId is the ID of the asset that has been parsed by SVGRenderer, and by the svgString line we don't know if we will use SVGRenderer. Do you mean cache the result of encodeDataURI if the asset doesn't need to go through SVGRenderer?

Reply:
I was just thinking that if `this.svgRendererAssetId === this.props.assetId` then the following must be true:
- we've already seen this asset once
- it passed all the checks at the time
- its ID hasn't changed, so presumably its content hasn't changed and would still pass all the checks
All of those things together imply to me that checking the ID can be used as a short-circuit to the cached result without checking the other things explicitly... unless one of my assumptions is bad, of course.